core/src/main/scripts/importer/validateData.py

```diff
@@ -1392,6 +1393,151 @@ def printDataInvalidStatement(self, value, col_index):
                    'column_number': col_index + 1,
                    'cause': value})
 
+class CustomNamespacesValidator(Validator):
+    """ Logic to validate custom namespace data."""
+
+    def __init__(self, *args, **kwargs):
+        super(CustomNamespacesValidator, self).__init__(*args, **kwargs)
+
+    def checkHeader(self, cols):
+        num_errors = super(CustomNamespacesValidator, self).checkHeader(cols)
+
+        #Namespaces column are compulsory when custom namespaces are defined in meta file
+        namespaces = []
+        if 'namespaces' in self.meta_dict:
+            namespaces = self.meta_dict['namespaces'].split(',')
+        for namespace in namespaces:
+            defined_namespace = namespace.strip().lower()
+            defined_namespace_found = any([True for col in cols if col.lower().startswith(defined_namespace + '.')])
+            if not defined_namespace_found:
+                self.logger.error('%s namespace defined but the file does not have any matching columns' 
+                                  % (defined_namespace))
+                num_errors += 1
+        
+        return num_errors
+
+class CNADiscreteLongValidator(CustomDriverAnnotationValidator, CustomNamespacesValidator):
+    """ Logic to validate discrete long CNA data."""
+    REQUIRED_HEADERS = [
+        'Sample_Id',
+        'Value'
+    ]
+    OPTIONAL_HEADERS = [
+        'Hugo_Symbol',
+        'Entrez_Gene_Id',
+        'cbp_driver',
+        'cbp_driver_annotation',
+        'cbp_driver_tiers',
+        'cbp_driver_tiers_annotation'
+    ]
+    REQUIRE_COLUMN_ORDER = False
+    ALLOW_BLANKS = True
+
+    ALLOWED_CNA_VALUES = ['-2', '-1.5', '-1', '0', '1', '2']
```

Comment:
`ALLOWED_CNA_VALUES` should includes `NULL_VALUES`, right? Because you are checking values in 1528, and `NA` is also an allowed value.

Reply:
It's also better to have some `NA` value in your test files, in that case, we can cover all possible allowed values.

Reply:
Thank you for rechecking this. I have added a test and made a fix to reallow `NA` values.